mypy/semanal.py

```diff
@@ -902,7 +902,9 @@ def analyze_base_classes(self, defn: ClassDef) -> None:
 
         base_types = []  # type: List[Instance]
         info = defn.info
-        for base_expr in defn.base_type_exprs:
+
+        base_type_exprs = self.check_with_metaclass(defn)
```

Comment:
Thanks! Much better now. I also change the API for check_with_metaclass() a bit, but it does exactly the same thing (except returns both values). Please review again.